nuxeo-drive-core/src/main/java/org/nuxeo/drive/listener/NuxeoDriveGroupUpdateListener.java

```diff
@@ -88,11 +91,19 @@ public void handleEvent(EventBundle events) {
         List<String> groupNames = new ArrayList<>();
         groupNames.add(groupName);
         // Get ancestor groups from the event context or compute them if not provided
-        List<String> ancestorGroups = (List<String>) context.getProperty(UserManagerImpl.ANCESTOR_GROUPS_PROPERTY_KEY);
-        if (ancestorGroups != null) {
-            groupNames.addAll(ancestorGroups);
-        } else {
-            groupNames.addAll(Framework.getService(UserManager.class).getAncestorGroups(groupName));
+        // and do it as system user in the local thread to access group directory
+        LoginStack loginStack = ClientLoginModule.getThreadLocalLogin();
+        loginStack.push(new SystemPrincipal(null), null, null);
```

Comment:
Why not do a `Framework.doPrivileged` instead?

Reply:
Also wondering why :smile:

Reply:
Needs to be investigated further, but it seems that `ClientLoginModule` is not initialized when using `Framework.doPrivileged` here => The principal is not retrieved when checking permissions on directories.